### ISSUE TITLE
Incorrect active account selection when there are multiple same accounts but with different authority

### ISSUE THREAD

**Issue by elepner** [AzureAD/microsoft-authentication-library-for-js#4601]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.22.0

### Wrapper Library

Not Applicable

### Wrapper Library Version

2.1.1

### Description

We've got multiple web applications using Azure B2C SSO and 2 login flows: SingInAndUp and JustSignIn. Therefore there are 2 accounts listed when calling ``getAllAccounts()``. Here's our example
```json
[
  {
    "homeAccountId": "590cd179-49b3-4266-828f-c0b3955a66ae-b2c_1_justsignin.81ff9595-acc4-480c-887d-cc1c10bd31e0",
    "environment": "rapidplanstaging.b2clogin.com",
    "tenantId": "",
    "username": "john.doe@invarion.com",
    "localAccountId": "590cd179-49b3-4266-828f-c0b3955a66ae",
    "name": "John Doe",
    "idTokenClaims": {
      "exp": 1647009520,
      "nbf": 1647005920,
      "ver": "1.0",
      "iss": "https://rapidplanstaging.b2clogin.com/81ff9595-acc4-480c-887d-cc1c10bd31e0/v2.0/",
      "sub": "590cd179-49b3-4266-828f-c0b3955a66ae",
      "aud": "eb3169e8-ff98-43f9-b02c-c91ef9006ab8",
      "nonce": "3c20ead3-986e-41ce-93d5-6a2cacc0e0d0",
      "iat": 1647005920,
      "auth_time": 1646923676,
      "oid": "590cd179-49b3-4266-828f-c0b3955a66ae",
      "name": "John Doe",
      "given_name": "John",
      "extension_InvarionId": "5afb1a56-758e-4f25-9580-b744ac7a5676",
      "family_name": "Doe",
      "emails": [
        "john.doe@invarion.com"
      ],
      "tfp": "B2C_1_JustSignIn",
      "at_hash": "ma7IEet5b2I7xs2SDpqRDw"
    }
  },
  {
    "homeAccountId": "590cd179-49b3-4266-828f-c0b3955a66ae-b2c_1_singinandup.81ff9595-acc4-480c-887d-cc1c10bd31e0",
    "environment": "rapidplanstaging.b2clogin.com",
    "tenantId": "",
    "username": "john.doe@invarion.com",
    "localAccountId": "590cd179-49b3-4266-828f-c0b3955a66ae",
    "name": "John Doe",
    "idTokenClaims": {
      "exp": 1647247524,
      "nbf": 1647243924,
      "ver": "1.0",
      "iss": "https://rapidplanstaging.b2clogin.com/81ff9595-acc4-480c-887d-cc1c10bd31e0/v2.0/",
      "sub": "590cd179-49b3-4266-828f-c0b3955a66ae",
      "aud": "eb3169e8-ff98-43f9-b02c-c91ef9006ab8",
      "nonce": "95b4c656-9d39-413e-b9f0-357b74e96848",
      "iat": 1647243924,
      "auth_time": 1647243923,
      "oid": "590cd179-49b3-4266-828f-c0b3955a66ae",
      "given_name": "John",
      "family_name": "Doe",
      "extension_InvarionId": "5afb1a56-758e-4f25-9580-b744ac7a5676",
      "name": "John Doe",
      "extension_CompanyId": "fa11a1d9-f8b0-4d04-a16c-29118cfa660f",
      "emails": [
        "john.doe@invarion.com"
      ],
      "tfp": "B2C_1_SingInAndUp",
      "at_hash": "428EFEgC_luMkcIbZZVNSA"
    }
  }
]
```
As you can see they have identical ``localAccountId`` which is picked up by ``getAccountInfoByFilter`` ([code](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/305422535d93dfe5ff4fb1246eb5e9dde939b985/lib/msal-browser/src/cache/BrowserCacheManager.ts#L429)) and picks up the first record which is incorrect when web app wants to use ``B2C_1_SingInAndUp`` flow.
Then it causes error during token refresh when calling ``POST: https://rapidplanstaging.b2clogin.com/rapidplanstaging.onmicrosoft.com/b2c_1_singinandup/oauth2/v2.0/token`` with message: "AADB2C90088: The provided grant has not been issued for this endpoint. Actual Value : B2C_1_SingInAndUp and Expected Value : B2C_1_JustSignIn" which, in turn, causes infinite loop between web app and azure login screen.

### Error Message

AADB2C90088: The provided grant has not been issued for this endpoint. Actual Value : B2C_1_SingInAndUp and Expected Value : B2C_1_JustSignIn

core.js:6210 ERROR Error: Uncaught (in promise): BrowserAuthError: interaction_in_progress: Interaction is currently in progress. Please ensure that this interaction has been completed before calling an interactive API.  For more visit: aka.ms/msaljs/browser-errors.
BrowserAuthError: interaction_in_progress: Interaction is currently in progress. Please ensure that this interaction has been completed before calling an interactive API.  For more visit: aka.ms/msaljs/browser-errors.
    at BrowserAuthError.AuthError [as constructor] (AuthError.js:27:1)
    at new BrowserAuthError (BrowserAuthError.js:169:1)
    at Function.push.vgJv.BrowserAuthError.createInteractionInProgressError (BrowserAuthError.js:236:1)
    at BrowserCacheManager.push.YBFq.BrowserCacheManager.setInteractionInProgress (BrowserCacheManager.js:813:39)
    at PublicClientApplication.push.nFfJ.ClientApplication.preflightInteractiveRequest (ClientApplication.js:540:1)
    at PublicClientApplication.push.nFfJ.ClientApplication.preflightBrowserEnvironmentCheck (ClientApplication.js:527:1)
    at PublicClientApplication.<anonymous> (ClientApplication.js:156:1)
    at step (_tslib.js:75:1)
    at Object.next (_tslib.js:56:45)
    at _tslib.js:49:1
    at resolvePromise (zone-evergreen.js:1213:1)
    at new ZoneAwarePromise (zone-evergreen.js:1390:1)
    at __awaiter (_tslib.js:45:1)
    at PublicClientApplication.push.nFfJ.ClientApplication.acquireTokenRedirect (ClientApplication.js:150:25)
    at MsalService.acquireTokenRedirect (azure-msal-angular.js:42:20)
    at MsalInterceptor.acquireTokenInteractively (azure-msal-angular.js:390:1)
    at CatchSubscriber.selector (azure-msal-angular.js:362:1)
    at CatchSubscriber.error (catchError.js:27:1)
    at subscribeToPromise.js:8:6
    at ZoneDelegate.invoke (zone-evergreen.js:372:1)

### Msal Logs

MSAL Logging:  [Mon, 14 Mar 2022 08:48:44 GMT] : @azure/msal-angular@2.1.1 : Info - Interceptor - 1 scopes found for endpoint
MSAL Logging:  [Mon, 14 Mar 2022 08:48:47 GMT] : @azure/msal-angular@2.1.1 : Error - Interceptor - acquireTokenSilent rejected with error. Invoking interaction to resolve.

### MSAL Configuration

```javascript
{
  "auth": {
    "clientId": "eb3169e8-ff98-43f9-b02c-c91ef9006ab8",
    "authority": "https://RapidPlanStaging.b2clogin.com/RapidPlanStaging.onmicrosoft.com/B2C_1_SingInAndUp",
    "knownAuthorities": [
      "https://RapidPlanStaging.b2clogin.com/RapidPlanStaging.onmicrosoft.com/B2C_1_SingInAndUp",
      "https://RapidPlanStaging.b2clogin.com/RapidPlanStaging.onmicrosoft.com/B2C_1_SingInAndUp"
    ],
    "redirectUri": "http://localhost:4200/login-redirect",
    "postLogoutRedirectUri": "http://localhost:4200/logged-out",
    "navigateToLoginRequestUrl": true
  },
  "cache": {
    "cacheLocation": "localStorage",
    "storeAuthStateInCookie": false
  },
  "system": {
    "loggerOptions": {
      "piiLoggingEnabled": true,
      "logLevel": 0
    }
  }
}
```

### Relevant Code Snippets

```javascript
Necessary information has been provided in the issue description
```

### Reproduction Steps

1. Have 2 web apps requiring different user flows
2. Login on the first one
3. Login on the second one
4. The 2nd web app would pick up the first account as active one

### Expected Behavior

The library should respect required authority provided in the configuration when picking up active account.

### Identity Provider

Azure B2C Basic Policy

### Browsers Affected (Select all that apply)

Chrome, Firefox, Edge, Safari, Internet Explorer

### Source

External (Customer)

**Comment by samuelkubai** [comment]
Hi @elepner, 

This is a confirmed bug and we have it in our backlog to fix and allow for you to possible select the active account based on both the local account ID and home account id.

I will keep you updated on the progress of this.

**Comment by sameerag** [comment]
Can we link the PR here once we have it up in github? cc @samuelkubai

**Comment by ghost** [comment]
:tada:This issue was addressed in #5004, which has now been successfully released as `@azure/msal-common@v7.2.0`.:tada:

We recommend upgrading to the latest version of `@azure/msal-browser` or `@azure/msal-node` to take advantage of this change.

Handy links:
* [Release Notes](https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-common-v7.2.0)
* [Official NPM Release](https://www.npmjs.com/package/@azure/msal-common/v/7.2.0)
